eis_toolkit/spatial_analyses/cell_based_association.py

```diff
@@ -0,0 +1,511 @@
+
+    Allow to add data to an existing matrix. Mesh size and placement are not
+    altered. On or multiple columns are added to the CBA matrix based on
+    targeted shapes and/or attributes.
+
+    Args:
+        geodataframe: path to vector file.
+        target_attribut: <optional> Name of the targeted attribute. If
+            no attribute are specified, added data fo each cell are based on
+            the absence/presence intersection between cells and the shapes
+            within the selected vector file. If specified, attribute must be
+            categorical (list) and will generate as much columns as it has
+            unique values in <subset_of_target_attribut_values>
+            (dummification).
+            Note: if a file contain multiple target attribute, <add_layer> has
+            to be executed as many times as there are target attributes in this
+            file.
+        Name: <optional> Name of the coluumn to add to the matrix.
+        buffer: <optional> Allow the use of a buffer around shapes before
+            the intersection with CBA cells.Minimize border effects or allow
+            increasing positive samples (i.e. cells with mineralization).
+            Default: False; Value (CRS units): size of the buffer.
+        subset_of_target_attribut_values: <optionnel> list of categorical
+            values of interest within the targeted attributes. Allow filtering
+            subset of relevant values.
+
+    Returns:
+        CBA object is completed.
+    """
+
+    # Allows a non-mutable default value
+    if subset_of_target_attribute_values is None:
+        subset_of_target_attribute_values = []
+
+    # Prerequisite: the CBA grid must already have been initialized
+    assert cba is not None
+
+    # Reading the vector file
+    geodata = geodataframe.copy(deep=True)
+
+    # No buffer
+    if buffer is False:
+        # Recovery of the grid calculated at the initialization of the CBA
+        grid = gpd.GeoDataFrame(cba)
+
+        # Adding a column to the CBA
+        dummies, join_grid = prepare_grid(geodata, grid, column, subset_of_target_attribute_values)
+
+    # Buffer of specified value
+    else:
+        # Calculation of the buffer disks
+        added_buf = get_disc(cba, grid, buffer)
+
+        # Adding a column to the CBA
+        dummies, join_disc = prepare_disc(geodata, added_buf, column, subset_of_target_attribute_values)
+
+    # Application of the indicated name (otherwise filename is used)
+    if name is not None:
+        dummies.name = name
+
+    # Completion of the CBA object (values and names)
+    cba = cba.join(dummies).replace(np.nan, 0)
+    if column == "":
+        cba[dummies.name] = cba[dummies.name].map(int)
+
+    cba = cba.astype("int32", errors="ignore")
+
+    return cba
+
+
+def prepare_grid(
+    geodata: gpd.GeoDataFrame,
+    grid: gpd.GeoDataFrame,
+    target_attribut: str,
+    subset_of_target_attribut_values: list,
+) -> Union[list, gpd.GeoDataFrame]:
```

Comment:
This should be `Tuple` instead of `Union`. Furthermore, a `list` is not returned from the function, rather `pandas` objects:
```suggestion
) -> Tuple[Union[pd.DataFrame, pd.Series], gpd.GeoDataFrame]:
```